File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       Map<Integer,Integer> m = new HashMap();
       for(int i=0;i<nums.length;i++){
          int ans  =  target-nums[i];
          if(m.containsKey(ans)){
            return new int[] {m.get(ans),i};
          }else{
            m.put(nums[i],i);
          }
       }

       return new int[]{};
        
    }
}

File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> m =  new Dictionary<int,int>();
        for(int i=0; i<nums.Length;i++){
            int ans  =  target-nums[i];
            if(m.ContainsKey(ans)){
                return new int[]{m[ans],i};
            }else{
             m[nums[i]] = i;
            }
        }
        return new int[]{};

        
    }
}